Sources/Sentry/SentryNetworkTracker.m

```diff
@@ -265,6 +288,114 @@ - (void)urlSessionTask:(NSURLSessionTask *)sessionTask setState:(NSURLSessionTas
     SENTRY_LOG_DEBUG(@"SentryNetworkTracker finished HTTP span for sessionTask");
 }
 
+- (void)captureEvent:(NSURLSessionTask *)sessionTask
+{
+    NSHTTPURLResponse *myResponse = (NSHTTPURLResponse *)sessionTask.response;
+    NSNumber *responseStatusCode = [NSNumber numberWithLongLong:myResponse.statusCode];
+
+    if (!self.isCaptureFailedRequests || ![self containsStatusCode:myResponse.statusCode]) {
+        return;
+    }
+
+    if (![self isTargetMatch:sessionTask.currentRequest.URL
+                 withTargets:SentrySDK.options.failedRequestTargets]) {
+        return;
+    }
+
+    NSString *message = [NSString
+        stringWithFormat:@"HTTP Client Error with status code: %ld", (long)myResponse.statusCode];
+
+    SentryEvent *event = [[SentryEvent alloc] initWithLevel:kSentryLevelError];
+
+    SentryThreadInspector *threadInspector = SentrySDK.currentHub.getClient.threadInspector;
+    NSArray<SentryThread *> *threads = [threadInspector getCurrentThreads];
+
+    SentryException *sentryException = [[SentryException alloc] initWithValue:message
+                                                                         type:@"HTTP-ClientError"];
+    sentryException.mechanism =
+        [[SentryMechanism alloc] initWithType:@"SentryNetworkTrackingIntegration"];
+
+    if (threads.count > 0) {
```

Comment:
`l`: We can remove the if check here. I the threads are empty, the for loop will do nothing.